src/dft/descriptor.cxx

```diff
@@ -37,7 +40,19 @@ void compute_default_strides(const std::vector<std::int64_t>& dimensions,
         strides[i] = strides[i + 1] * dimensions[i];
     }
     strides[0] = 0;
-    output_strides = strides;
+    if constexpr(dom == domain::COMPLEX){
+        output_strides = strides;
+    } else{
+        std::vector<std::int64_t> out_strides(rank + 1, 1);
+        if(rank > 2){
+            out_strides[2] = dimensions[i];
+        }
+        if(rank > 1){
+            out_strides[1] = out_strides[2] * (dimensions[1]/2 + 1);
+        }
+        out_strides[0] = 0;
+        output_strides = std::move(out_strides);
```

Comment:
```suggestion
        std::vector<std::int64_t> out_strides(rank + 1, 1);
        // I'm assuming rank >= 1
        out_strides[rank - 1] = out_strides[rank] * (dimensions[rank - 1]/2 + 1); // overwritten in the end if rank == 1
        for (int i = rank - 2; i >= 1; --i) {
            out_strides[i] = out_strides[i + 1] * dimensions[i];
        }
        out_strides[0] = 0; // offset index, am I right? 
        output_strides = std::move(out_strides);
```
I believe that's what you need, BUT is that chunk meant to cover both compute direction (bwd and fwd) for real transforms? It looks like it's designed for forward only... For backward transform, you'd need to swap input and output strides.
Also, for in-place transforms, there is a requirement for padding in the strides associated with the real domain: you basically need every real stride entry to be twice the corresponding complex stride entry.

Reply:
Exactly, this is for forward transform. I don't think we have the information at this point, whether the descriptor will be used for forward or backward and I wanted to cover at least one.